exla/lib/exla/defn.ex

```diff
@@ -318,7 +318,7 @@ defmodule EXLA.Defn do
         {computation, extra, hooks} =
           to_computation.(expr || fun.(vars), inputs_and_shapes, used_hooks)
 
-        executable = EXLA.Computation.compile(computation, client, shapes)
```

Comment:
Is there a way to write a test in which this makes a difference?

Reply:
Yeah, we can write a test that sets the option to an invalid device_id and see if it raises accordingly. :)

Reply:
Done!